include/seqan3/alignment/pairwise/detail/policy_alignment_matrix.hpp

```diff
@@ -0,0 +1,172 @@
+     * sequence sizes and the initial score value. In the banded alignment, the alignment matrix is reduced to
+     * the column count times the band size.
+     *
+     * ### Exception
+     *
+     * Might throw std::bad_alloc if the requested matrix size exceeds the available memory or
+     * seqan3::invalid_alignment_configuration if the band does not allow a valid computation of the
+     * configured alignment.
+     *
+     * \throws std::bad_alloc or seqan3::invalid_alignment_configuration
+     */
+    auto acquire_matrices(size_t const sequence1_size,
+                          size_t const sequence2_size,
+                          score_type initial_score = score_type{}) const
+    {
+        assert(sequence1_size < static_cast<uint64_t>(std::numeric_limits<int64_t>::max()));
+        assert(sequence2_size < static_cast<uint64_t>(std::numeric_limits<int64_t>::max()));
+
+        if constexpr (traits_t::is_banded)
+            check_valid_band_configuration(sequence1_size, sequence2_size);
+
+        static thread_local alignment_matrix_t alignment_matrix{};
+        static thread_local coordinate_matrix<uint32_t> index_matrix{};
+
+        // Increase dimension by one for the initialisation of the matrix.
+        size_t const column_count = sequence1_size + 1;
+        size_t row_count = sequence2_size + 1;
+
+        if constexpr (traits_t::is_banded)
+        {
+            assert(upper_diagonal - lower_diagonal + 1 > 0);
+
+            row_count = std::min<int64_t>(upper_diagonal - lower_diagonal + 1, row_count);
+        }
+
+        alignment_matrix.resize(column_index_type{column_count}, row_index_type{row_count}, initial_score);
+        alignment_matrix.resize(column_index_type{column_count}, row_index_type{row_count}, initial_score);
+
+        return std::tie(alignment_matrix, index_matrix);
+    }
+
+    /*!\brief Checks whether the band is valid for the given sequence sizes.
+     *
+     * \param[in] sequence1_size The size of the first sequence.
+     * \param[in] sequence2_size The size of the second sequence.
+     *
+     * \throws seqan3::invalid_alignment_configuration if the band is invalid for the given sequence sizes and the
+     *         alignment configuration.
+     */
+    void check_valid_band_configuration(size_t const sequence1_size, size_t const sequence2_size) const
+    {
+        bool const upper_diagonal_ends_before_last_cell = (upper_diagonal + sequence2_size) < sequence1_size;
+        bool const lower_diagonal_ends_behind_last_cell = (-lower_diagonal + sequence1_size) < sequence2_size;
+
+        if (upper_diagonal_ends_before_last_cell || lower_diagonal_ends_behind_last_cell)
+            throw invalid_alignment_configuration{"The selected band [" + std::to_string(lower_diagonal) + ":" +
+                                                  std::to_string(upper_diagonal) + "] does not cover the last cell."};
```

Comment:
The same here, only "validity checks" that concern global alignment, wrapped in `if constexpr global alignment`